extensions/rke1/nodetemplates/nodetemplates.go

```diff
@@ -39,3 +47,66 @@ type NodeTemplate struct {
 	UUID                            string                           `json:"uuid,omitempty" yaml:"uuid,omitempty"`
 	UseInternalIPAddress            *bool                            `json:"useInternalIpAddress,omitempty" yaml:"useInternalIpAddress,omitempty"`
 }
+
+func providerTemplateConfigKeys() []string {
+	return []string{
+		AmazonEC2NodeTemplateConfigurationFileKey,
+		AzureNodeTemplateConfigurationFileKey,
+		HarvesterNodeTemplateConfigurationFileKey,
+		LinodeNodeTemplateConfigurationFileKey,
+		VmwareVsphereNodeTemplateConfigurationFileKey,
+	}
+}
+
+func (n1 *NodeTemplate) MergeOverride(n2 *NodeTemplate, providerTemplateConfigKey string) (*NodeTemplate, error) {
+	var n1Data map[string]any
+	n1YAML, err := yaml.Marshal(&n1)
+	if err != nil {
+		return nil, errors.Wrap(err, "MergeOverride: ")
+	}
+	err = yaml.Unmarshal(n1YAML, &n1Data)
+	if err != nil {
+		return nil, errors.Wrap(err, "MergeOverride: ")
+	}
+
+	var n2Data map[string]any
+	tempConfigYAML, err := yaml.Marshal(&n2)
```

Comment:
Just a small nit here, the other one is called `n1YAML`, can we make this one `n2YAML` or the other one `actualYaml`, or something you think can unify these two?